src/sentry/lang/javascript/processing.py

```diff
@@ -0,0 +1,146 @@
+        new_frame["pre_context"] = symbolicated["pre_context"]
+    if symbolicated.get("context_line"):
+        new_frame["context_line"] = symbolicated["context_line"]
+    if symbolicated.get("post_context"):
+        new_frame["post_context"] = symbolicated["post_context"]
+    if symbolicated.get("status"):
+        frame_meta = new_frame.setdefault("data", {})
+        frame_meta["symbolicator_status"] = symbolicated["status"]
+
+    return new_frame
+
+
+# TODO: Change this error handling to be JS-specific?
+def _handle_response_status(event_data, response_json):
+    if not response_json:
+        error = SymbolicationFailed(type=EventError.NATIVE_INTERNAL_FAILURE)
+    elif response_json["status"] == "completed":
+        return True
+    elif response_json["status"] == "failed":
+        error = SymbolicationFailed(
+            message=response_json.get("message") or None, type=EventError.NATIVE_SYMBOLICATOR_FAILED
+        )
+    else:
+        logger.error("Unexpected symbolicator status: %s", response_json["status"])
+        error = SymbolicationFailed(type=EventError.NATIVE_INTERNAL_FAILURE)
+
+    write_error(error, event_data)
+
+
+def get_frames_for_symbolication(frames, data):
+    return [dict(frame) for frame in reversed(frames)]
+
+
+def is_sourcemap_image(image):
+    return (
+        bool(image)
+        and image.get("type") == "sourcemap"
+        and image.get("debug_id") is not None
+        and image.get("code_file") is not None
+    )
+
+
+def sourcemap_images_from_data(data):
+    return get_path(data, "debug_meta", "images", default=(), filter=is_sourcemap_image)
+
+
+def process_payload(data):
+    # We cannot symbolicate JS stacktraces without a release.
+    if data.get("release") is None:
+        return
+
+    project = Project.objects.get_from_cache(id=data["project"])
+
+    symbolicator = Symbolicator(project=project, event_id=data["event_id"], release=data["release"])
+
+    modules = sourcemap_images_from_data(data)
+
+    stacktrace_infos = find_stacktraces_in_data(data)
+    stacktraces = [
+        {
+            "frames": get_frames_for_symbolication(sinfo.stacktrace.get("frames") or (), data),
+        }
+        for sinfo in stacktrace_infos
+    ]
+
+    if not any(stacktrace["frames"] for stacktrace in stacktraces):
+        return
+
+    response = symbolicator.process_js(
+        stacktraces=stacktraces, modules=modules, dist=data.get("dist")
+    )
+
+    if not _handle_response_status(data, response):
+        return data
+
+    assert len(stacktraces) == len(response["stacktraces"]), (stacktraces, response)
+
+    for sinfo, raw_stacktrace, complete_stacktrace in zip(
+        stacktrace_infos, response["raw_stacktraces"], response["stacktraces"]
+    ):
+        new_frames = []
+        new_sinfo_frames = []
+
+        for sinfo_frame, raw_frame, complete_frame in zip(
+            sinfo.stacktrace["frames"], raw_stacktrace["frames"], complete_stacktrace["frames"]
+        ):
```

Comment:
We need to `reverse` the raw and completed frames here, as the `sinfo.stacktrace` is in reverse order.

We are also lucky that the `get_frames_for_symbolication` does not filter frames. For native it does, and therefore merging back the symbolicated stack traces needs to consider the `original_index`.

Reply:
It did filter them, and after discussion with Sebastian, we dropped the filter for this exact reason 😅